web/themes/new_weather_theme/assets/js/components/forecast-details-toggle.js

```diff
@@ -0,0 +1,20 @@
+Array.from(document.querySelectorAll(".wx-forecast-details-toggle"))
+  .forEach(toggleContainer => {
+    // The click event for each toggle button will
+    // update the visual selection class and also update the
+    // selection data attribute
+    const buttons = Array.from(toggleContainer.querySelectorAll("button"));
+    buttons.forEach(button => {
+      button.addEventListener("click", () => {
+        const selectedName = button.dataset.selectionName;
+        toggleContainer.setAttribute("data-selected", button.dataset.selectionName);
+        buttons.forEach(btn => {
+          if(btn.dataset.selectionName === selectedName){
+            btn.classList.remove("usa-button--outline");
+          } else {
+            btn.classList.add("usa-button--outline");
+          }
+        });
+      });
+    });
+  });
```

Comment:
This is very much a take-it-or-leave-it suggestion. It kind of evolved as I was thinking about it, so I'll lay out my thinking.

1. Attaching an event handler to a closure created inside this loop means we'll have a unique instance of the function for every button, or 14 instances of the function. It's not a lot, but it's still extra memory. This particular optimization is one that lives at the front of my brain because in Windows applications, it's a prime source of memory leaks. Not really an issue here, though, since the button's lifetime is the window's lifetime. 🤷‍♂️ 
2. One downside to moving to a single event handler is having to recompute the toggle and list of buttons each time. But it turns out `getElementsByTagName` and `getElementsByClassName` are at least one order of magnitude faster than `querySelector*`. TIL! So if these methods are sufficiently fast, then any performance hit is probably totally erased by not keeping the list of button nodes in memory forever, too, since they were being captured by the event handler closure.

But I don't know if this is actually significant enough to matter to anything.

```suggestion
// The click event for each toggle button will
// update the visual selection class and also update the
// selection data attribute
const toggleHandler = ({ target: button }) => {
  const toggleContainer = button?.closest(".wx-forecast-details-toggle");
  const selectedName = button?.dataset.selectionName;

  toggleContainer?.setAttribute("data-selected", selectedName);

  const buttons = Array.from(toggleContainer.getElementsByTagName("button"));
  buttons.forEach((btn) => {
    btn.classList.add("usa-button--outline");
    if (btn.dataset.selectionName === selectedName) {
      btn.classList.remove("usa-button--outline");
    }
  });
};

Array.from(document.getElementsByClassName("wx-forecast-details-toggle"))
  .forEach(toggleContainer => {
    const buttons = Array.from(toggleContainer.getElementsByTagName("button"));
    buttons.forEach(button => {
      button.addEventListener("click", toggleHandler);
    });
  });
```